packages/ui/src/components/Menu/Item.tsx

```diff
@@ -4,96 +4,75 @@ import type { MouseEventHandler, ReactNode, Ref } from 'react'
+}) => `
   display: inline-block;
-  font-size: ${({ theme }) => theme.typography.bodySmall.fontSize};
-  line-height: ${({ theme }) => theme.typography.bodySmall.lineHeight};
+  font-size: ${theme.typography.bodySmall.fontSize};
+  line-height: ${theme.typography.bodySmall.lineHeight};
   font-weight: inherit;
-  padding: ${({ theme }) => `${theme.space['0.5']} ${theme.space['1']}`};
-  color: ${({ theme }) => theme.colors.neutral.text};
-  border: 0;
-  border-bottom: 1px solid ${({ theme }) => theme.colors.neutral.border};
+  padding: ${`${theme.space['0.5']} ${theme.space['1']}`};
+  border: none;
+  ${
+    borderless ? '' : `border-bottom: 1px solid ${theme.colors.neutral.border};`
+  }
   cursor: pointer;
-  transition: color 300ms;
   min-width: 110px;
-  background-color: transparent;
-  border-radius: ${({ theme }) => theme.radii.none};
-  &:hover,
-  &:focus {
-    color: ${({ theme }) => theme.colors.primary.textHover};
-    svg {
-      transition: fill 300ms;
-      fill: ${({ theme }) => theme.colors.primary.textHover};
-    }
-  }
 
-  &:last-child {
-    border-bottom: 0;
-    border-radius: 0 0
-      ${({ theme }) => ` ${theme.radii.default} ${theme.radii.default}`};
+  color: ${theme.colors[sentiment][disabled ? 'textDisabled' : 'text']};
+  svg {
+    fill: ${theme.colors[sentiment][disabled ? 'textDisabled' : 'text']};
   }
 
-  ${({ itemVariant, theme }) =>
-    itemVariant ? variantStyle[itemVariant]?.(theme) : null}
-
-  ${({ disabled, theme }) =>
+  ${
     disabled
       ? `
-    cursor: not-allowed;
-    color: ${theme.colors.neutral.textDisabled};
-
-    &:hover,
-    &:focus {
-      color: ${theme.colors.neutral.textHover};
-      background-color: ${theme.colors.neutral.backgroundWeak};
-      svg {
-        fill: ${theme.colors.neutral.textHover};
-      }
-    }
+        cursor: not-allowed;
+      `
+      : `
+          &:hover,
+          &:focus {
+            color: ${theme.colors[sentiment].textHover};
+            svg {
+              fill: ${theme.colors[sentiment].textHover};
+            }
+          }`
+  }
   `
-      : null}
 
-  ${({ borderless }) => (borderless ? `border: 0;` : null)}
+const StyledItem = styled('button', {
+  shouldForwardProp: prop => !['borderless', 'sentiment'].includes(prop),
+})<{
+  borderless: boolean
+  disabled: boolean
+  sentiment: MenuItemSentiment
+}>`
+  ${({ theme, borderless, sentiment, disabled }) =>
+    itemCoreStyle({ theme, borderless, sentiment, disabled })}
+  background: none;
 `
-const StyledLinkItem = styled(StyledItem)`
-  text-decoration: none;
-`.withComponent('a')
 
-type VariantItem = keyof typeof variantStyle
+const StyledLinkItem = styled('a', {
+  shouldForwardProp: prop => !['borderless', 'sentiment'].includes(prop),
+})<{
+  borderless: boolean
+  disabled: boolean
+  sentiment: MenuItemSentiment
+}>`
+  ${({ theme, borderless, sentiment, disabled }) =>
+    itemCoreStyle({ theme, borderless, sentiment, disabled })}
+  text-decoration: none;
```

Comment:
Possible to add this:

```suggestion
  text-decoration: none;
  
  &:focus {
    text-decoration: none;
  }
```